src/librustc_target/abi/mod.rs

```diff
@@ -1081,4 +1119,57 @@ impl<'a, Ty> TyLayout<'a, Ty> {
             Abi::Aggregate { sized } => sized && self.size.bytes() == 0
         }
     }
+
+    /// Determines if zero-initializing this type might be okay.
+    /// This is conservative: in doubt, it will answer `true`.
+    pub fn might_permit_zero_init<C, E>(
+        &self,
```

Comment:
Note that other methods don't take `self` by reference. This would also let you mutate `self` (if you need to e.g. use `for_variant`).

Reply:
Eh, this is next to `is_unsized` and `is_zst` which both take it by reference. I am confused by your statement.

Reply:
I was looking above, didn't see the other methods below. Point still stands, there's no reason to ever pass types like `TyLayout` by reference.

Reply:
I changed my new method to `self`. Changing the others leads to errors in various places as `TyLayout` is not in general known to be `Copy`.